docs/map.rst

```diff
@@ -0,0 +1,629 @@
+
+There are several other more complex mapping functions available that let you
+look at other parts of the entry (such as they key) when transforming the
+value. For the full list see the `API documentation
+<https://hackage.haskell.org/package/containers-0.5.10.2/docs/Data-Map-Strict.html#g:15>`_.
+
+
+Set-like Operations
+^^^^^^^^^^^^^^^^^^^
+
+.. _union:
+
+Union
+"""""
+
+::
+
+    Map.union :: Ord k => Map k v -> Map k v -> Map k v
+    Map.union l r = ...
+
+:map:`union` returns a map containing all entries that are keyed in either of
+the two map. If the same key appears in both maps, the value from the left map
+``l`` taken (`set union <https://en.wikipedia.org/wiki/Union_(set_theory)>`_).
+
+::
+
+    Map.union Map.empty (Map.fromList [(1,"one"),(2,"two")])
+    > fromList [(1,"one"),(2,"two")]
+
+    Map.union (Map.fromList [(1, "uno")]) (Map.fromList [(1,"one"),(2,"two")])
+    > fromList [(1,"uno"),(2,"two")]
+
+Intersection
+""""""""""""
+
+::
+
+    Map.intersection :: Ord k => Map k v -> Map k v -> Map k v
+    Map.intersection l r = ...
+
+:map:`intersection` returns a map containing all entries that are keyed in both
+maps ``l`` and ``r``. The value from the left map is taken if the key exists in
+both maps (`set intersection
+<https://en.wikipedia.org/wiki/Intersection_(set_theory)>`_).
+
+::
+
+    Map.intersection Map.empty (Map.fromList [(1,"one"), (2,"two")])
+    > fromList []
+
+    Map.intersection (Map.fromList [(1, "uno")]) (Map.fromList [(1,"one"),(2,"two")])
+    > fromList [(1,"uno")]
+
+Difference
+""""""""""
+
+::
+
+    Map.difference :: Ord k => Map k v -> Map k v -> Map k v
+    Map.difference l r = ...
+
+:map:`difference` returns a map containing all entries that are keyed in the
+``l`` map but not the ``r`` map (`set difference/relative compliment
+<https://en.wikipedia.org/wiki/Complement_(set_theory)#Relative_complement>`_).
+
+::
+
+    Map.difference (Map.fromList [(1,"one"), (2,"two"), (3,"three")]) Map.empty
+    > fromList [(1,"uno"),(2,"two"),(3,"three")]
+
+    Map.difference (Map.fromList[(1,"one"), (2,"two")]) (Map.fromList [(1,"uno")])
+    > fromList [(2,"two")]
+
+Subset (submap)
+"""""""""""""""
+
+::
+
+    Map.isSubmapOf :: (Eq a, Ord k) => Map k v -> Map k v -> Bool
+    Map.isSubmapOf l r = ...
+
+:map:`isSubmapOf` returns ``True`` if all entries--(keys, value) pairs--in the
+left map ``l`` exist in the right map ``r``, ``False`` otherwise (`subset
+<https://en.wikipedia.org/wiki/Subset>`_).
+
+.. NOTE::
+   We use `infix notation
+   <https://wiki.haskell.org/Infix_operator#Using_infix_functions_with_prefix_notation>`_
+   so that it reads nicer. These are back-ticks (`), not quotes.
+
+::
+
+    Map.empty `Map.isSubmapOf ` Map.empty
+    > True
+
+    Map.empty `Map.isSubmapOf` (Map.fromList [(1,"one"), (2,"two")])
+    > True
+
+    (Map.singleton 1 "uno") `Map.isSubmapOf` (Map.fromList [(1,"one"), (2,"two")])
+    > True
+
+
+Typeclass Instances
+-------------------
+
```

Comment:
Listing these without explaining their importance, relevance, or specific meaning for maps seems kind of silly to me. I think it could be helpful to explain how to use the `Foldable` and `Traversable` instances, but most of the rest of this is pretty boring.

Reply:
I think you're right, I don't think these instances really add anything so I've removed the section from all the pages.

Reply:
Also, you probably don't want to use the `Foldable` or `Traversable` instances for maps at all; if you want to fold or traverse them you probably should use `fold*WithKey` or `traverseWithKey`. The `WithKey` folds and traversals would probably be worth discussing.

Reply:
Removed entirely.